cytoolz/itertoolz.pyx

```diff
@@ -959,8 +959,11 @@ cdef class sliding_window:
     """
     def __cinit__(self, Py_ssize_t n, object seq):
         cdef Py_ssize_t i
+        cdef object none = None
         self.iterseq = iter(seq)
         self.prev = PyTuple_New(n)
+        Py_INCREF(none)
+        PyTuple_SET_ITEM(self.prev, 0, none)
```

Comment:
The original `self.prev` never makes it to Python code, in `__next__` a new tuple is created which is what actually makes it to Python. I am not sure where the bug is.